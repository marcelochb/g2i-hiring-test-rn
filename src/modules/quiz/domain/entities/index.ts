export {QuizEntity} from './quiz_entity'
