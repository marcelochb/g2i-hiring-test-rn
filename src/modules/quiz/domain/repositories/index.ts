export * from './quiz_repository_interface';
