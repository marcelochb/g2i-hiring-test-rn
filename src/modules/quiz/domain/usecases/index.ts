export * from './get_all_quiz_usecase';
