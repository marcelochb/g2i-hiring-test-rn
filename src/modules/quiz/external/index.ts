export * from './datasources/quiz_datasource';
export * from './mocks/quiz_mock'
