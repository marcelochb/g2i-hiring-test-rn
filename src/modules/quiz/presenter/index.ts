export * from './home';
export * from './question';
export * from './quiz_binds';
