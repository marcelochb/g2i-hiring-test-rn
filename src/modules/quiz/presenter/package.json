{
  "name": "@src/modules/quiz/presenter",
  "version": "0.0.1"
}
