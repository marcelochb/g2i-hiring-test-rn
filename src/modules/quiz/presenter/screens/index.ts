export * from './home';
export * from './question';
export * from './result';
