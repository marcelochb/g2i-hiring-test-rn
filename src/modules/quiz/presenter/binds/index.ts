import { InjectContants } from "@src/core/constants"
import { ApiClient } from "@src/core/services"
import { container } from "tsyringe"
import { GetAllQuizUseCase } from "../../domain/usecases/get_all_quiz_usecase";
import { QuizDatasource } from "../../external/datasources/quiz_datasource";
import { QuizRepository } from "../../infra/repositories";

export const quizDependences = () => {
  container.register(InjectContants.IApiClient,{useValue: new ApiClient()});
  container.register(InjectContants.IQuizDatasource,{useClass: QuizDatasource});
  container.register(InjectContants.IQuizRepository,{useClass: QuizRepository});
  container.register(InjectContants.GetAllQuizUseCase,{useClass: GetAllQuizUseCase});
}
