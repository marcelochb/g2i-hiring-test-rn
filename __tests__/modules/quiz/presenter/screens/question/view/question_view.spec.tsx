

import 'reflect-metadata';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {create} from 'react-test-renderer';
import { QuizEntity } from '@src/modules/quiz/domain';
import { IQuizReducer, Question } from '@src/modules/quiz/presenter';
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: jest.fn(),
}));
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn().mockImplementation(() => jest.fn()),
  useSelector: jest.fn().mockImplementation(callback => {
    return callback({quizReducer: {
      loading: false,
      error: false,
      empty: false,
      quizzes: [{category: 'teste',question: 'teste', correct_answer: true}],
      currentQuiz: {category: 'teste',question: 'teste', correct_answer: true} as IQuizReducer,
      totalCount: 1,
      correctAnswers: 0
    }
    });
  })
}));

describe('Question view', () => {
  test('When it loaded, should be returned current question', () => {
    const quizzes = [new QuizEntity({category:'Technology',correct_answer:false,question:'What is it?'})];
    const store = configureMockStore([])({quizzes});
    const component = create(
      <Provider store={store}>
        <Question />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
