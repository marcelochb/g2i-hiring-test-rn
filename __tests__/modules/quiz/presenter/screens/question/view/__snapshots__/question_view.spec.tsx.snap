// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question view When it loaded, should be returned current question 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 0,
      }
    }
  />,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0D0D0D",
              "fontFamily": "Roboto-Bold",
              "fontSize": 20,
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        teste
      </Text>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#0D0D0D",
                "borderWidth": 1,
                "padding": 16,
              },
              Object {
                "marginBottom": 16,
                "marginHorizontal": 8,
                "padding": 32,
                "paddingVertical": 32,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#0D0D0D",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 20,
                },
                Object {
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            teste
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-evenly",
                "marginTop": 32,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#0D0D0D",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 20,
                    },
                    Object {},
                    undefined,
                  ]
                }
              >
                TRUE
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#0D0D0D",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 20,
                    },
                    Object {},
                    undefined,
                  ]
                }
              >
                FALSE
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#0D0D0D",
                "fontFamily": "Roboto-Regular",
                "fontSize": 20,
              },
              Object {
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          undefined of 1
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>,
]
`;
